api/errors/errors.go

```diff
@@ -1,47 +1,57 @@
 package errors
 
-import "net/http"
+// NotFoundError signals that the requested object doesn't exist
+type NotFoundError interface {
+	error
+	NotFound()
+}
 
-// apiError is an error wrapper that also
-// holds information about response status codes.
-type apiError struct {
+// InvalidParameterError signals that the user input is invalid
+type InvalidParameterError interface {
```

Comment:
Rather than export these interfaces for doing tests, it might be better to export an `IsInvalidParameter` function. This ensures that causality detection is taken into account and keeps that logic in a single place. Otherwise, users of this package will have to use a combination of `Cause()` detection and recursive detection of this specific interface.